fleet/fleet.go

```diff
@@ -131,18 +132,15 @@ type Fleet interface {
 func NewFleet(config Config) (Fleet, error) {
 	var trans http.RoundTripper
 
-	switch config.Endpoint.Scheme {
-	case "unix", "file":
-		if config.Endpoint.Host != "" {
-			// This commonly happens if the user misses the leading slash after the
-			// scheme. For example, "unix://var/run/fleet.sock" would be parsed as
-			// host "var".
-			return nil, maskAnyf(invalidEndpointError, "cannot connect to host %q with scheme %q", config.Endpoint.Host, config.Endpoint.Scheme)
+	tunneling := config.Tunnel.Tunnel != ""
+	// If a tunnel is provided we need to overwrite the http.Transport.Dial function
+	// to use the tunnel
+	if tunneling {
```

Comment:
@xh3b4sd not sure if this is better :/ I dislike the code replication with 156 ff. What do you think?